teuthology/task/selinux.py

```diff
@@ -141,7 +141,7 @@ def get_denials(self):
         se_allowlist = self.config.get('allowlist', [])
         if se_allowlist:
             known_denials.extend(se_allowlist)
-        ignore_known_denials = r'\'\(' + str.join(r'\|', known_denials) + r'\)\''
+        ignore_known_denials = r"""'\(' + str.join(r'\|', known_denials) + r'\)'"""
```

Comment:
This results in:
```2023-01-30T19:17:22.557 INFO:teuthology.orchestra.run.smithi170.stderr:bash: -c: line 0: `sudo grep -a 'avc: .*denied' /var/log/audit/audit.log | grep -av '\(' + str.join(r'\|', known_denials) + r'\)''```
notice we're grepping for the literal `"known_denials"`, not the actual list of strings

Reply:
Hah, will fix.

Reply:
I'm also testing a different approach at https://github.com/ceph/teuthology/pull/1810, which eliminates the need to escape.

Reply:
Ah, feel free to close this one if you prefer that one.  Either way, let's run a small suite prior to merge though.

Reply:
I'm rerunning the same suite you'd scheduled; we'll see if it passes. I'd thought of that approach because it'd be far easier to unit-test, but to me it sort of feels like there won't be much to test in that case as we wouldn't be generating a confusing-looking escaped string.

Reply:
Yep, works for me.  https://github.com/ceph/teuthology/pull/1810 seems cleaner.

Reply:
Update the Tracker with whatever PR ends up being used